src/query_results.js

```diff
@@ -204,221 +204,220 @@ module.exports = class QueryResult {
-    let initialQueryEdgeID, initialQueryNodeIDToMatch;
-    toPairs(dataByEdge).some(([queryEdgeID, {connected_to, records}]) => {
-      const inputQueryNodeID = helper._getInputQueryNodeID(records[0]);
-      const outputQueryNodeID = helper._getOutputQueryNodeID(records[0]);
+    let initialQEdgeID, initialQNodeIDToMatch;
+    toPairs(recordsByQEdgeID).some(([queryEdgeID, {connected_to, records}]) => {
+      const inputQNodeID = helper._getInputQueryNodeID(records[0]);
+      const outputQNodeID = helper._getOutputQueryNodeID(records[0]);
 
       if (connected_to.length === 0) {
-        initialQueryEdgeID = queryEdgeID;
-        initialQueryNodeIDToMatch = inputQueryNodeID;
+        initialQEdgeID = queryEdgeID;
+        initialQNodeIDToMatch = inputQNodeID;
       } else {
         connected_to.some((c) => {
-          const nextEdge = dataByEdge[c];
-          const inputQueryNodeID1 = helper._getInputQueryNodeID(nextEdge.records[0]);
-          const outputQueryNodeID1 = helper._getOutputQueryNodeID(nextEdge.records[0]);
-          if (!initialQueryEdgeID) {
-            if ([inputQueryNodeID1, outputQueryNodeID1].indexOf(inputQueryNodeID) === -1) {
-              initialQueryEdgeID = queryEdgeID;
-              initialQueryNodeIDToMatch = inputQueryNodeID;
+          const nextEdge = recordsByQEdgeID[c];
+          const inputQNodeID_1 = helper._getInputQueryNodeID(nextEdge.records[0]);
+          const outputQNodeID_1 = helper._getOutputQueryNodeID(nextEdge.records[0]);
+          if (!initialQEdgeID) {
+            if ([inputQNodeID_1, outputQNodeID_1].indexOf(inputQNodeID) === -1) {
+              initialQEdgeID = queryEdgeID;
+              initialQNodeIDToMatch = inputQNodeID;
 
               // like calling break in a loop
               return true;
-            } else if ([outputQueryNodeID1, outputQueryNodeID1].indexOf(outputQueryNodeID) === -1) {
-              initialQueryEdgeID = queryEdgeID;
-              initialQueryNodeIDToMatch = outputQueryNodeID;
+            } else if ([outputQNodeID_1, outputQNodeID_1].indexOf(outputQNodeID) === -1) {
+              initialQEdgeID = queryEdgeID;
+              initialQNodeIDToMatch = outputQNodeID;
 
               // like calling break in a loop
               return true;
             }
           }
         });
 
-        if (initialQueryEdgeID) {
+        if (initialQEdgeID) {
           // like calling break in a loop
           return true;
         }
       }
     });
 
-    debug(`initialQueryEdgeID: ${initialQueryEdgeID}, initialQueryNodeIDToMatch: ${initialQueryNodeIDToMatch}`);
+    debug(`initialQEdgeID: ${initialQEdgeID}, initialQNodeIDToMatch: ${initialQNodeIDToMatch}`);
 
-    // 'preresult' just means it has the data needed to assemble a result,
+    // 'queryGraphSolution' just means it has the data needed to assemble a result,
     // but it's formatted differently for easier pre-processing.
```

Comment:
This comment was specific to the term `preresult`, so we could delete these two lines